File: VB/T292798/Form1.vb
Imports System
Imports System.ComponentModel
Imports System.Drawing
Imports DevExpress.DataAccess.UI.Sql

Namespace T292798

    Public Partial Class Form1
        Inherits DevExpress.XtraEditors.XtraForm

        Public Sub New()
            InitializeComponent()
        End Sub

        Private Sub Form1_Load(ByVal sender As Object, ByVal e As EventArgs)
            sqlDataSource1.Fill()
        End Sub

        Private Sub btConfigureConnection_Click(ByVal sender As Object, ByVal e As EventArgs)
            sqlDataSource1.ConfigureConnection
        End Sub

        Private Sub btEditQuery_Click(ByVal sender As Object, ByVal e As EventArgs)
            sqlDataSource1.Queries("DefaultQuery").EditQuery()
            sqlDataSource1.Fill()
            gridView1.PopulateColumns()
        End Sub
    End Class
End Namespace

File: CS/T292798/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Windows.Forms;
using DevExpress.DataAccess.UI.Sql;

namespace T292798 {
    public partial class Form1 : DevExpress.XtraEditors.XtraForm {
        public Form1() {
            InitializeComponent();            
        }

        private void Form1_Load(object sender, EventArgs e) {
            sqlDataSource1.Fill();
        }

        private void btConfigureConnection_Click(object sender, EventArgs e) {
            SqlDataSourceUIHelper.ConfigureConnection(sqlDataSource1);
        }

        private void btEditQuery_Click(object sender, EventArgs e) {
            SqlDataSourceUIHelper.EditQuery(sqlDataSource1.Queries["DefaultQuery"]);

            sqlDataSource1.Fill();

            gridView1.PopulateColumns();
        }
    }
}
